File: JSONWIRE/win.js
const {Builder, By, Key, until} = require('selenium-webdriver');

async function example() {
  const cap = {
    'browserName': 'chrome',
    'browserstack.local':'false',
    'browserstack.uploadMedia':["media://<FILE_HASHED_ID>"],
    'browser_version': 'latest',
    'os':'Windows',
    'os_version':'10'
  }
  let driver = await new Builder().usingServer('https://YOUR_USERNAME:YOUR_ACCESSKEY@hub-cloud.browserstack.com/wd/hub').
    withCapabilities(cap).build();
  try {
    await driver.get('http://www.fileconvoy.com');
    await driver.findElement(By.id('upfile_0')).sendKeys("C:\\Users\\hello\\Documents\\<MEDIA_FOLDER>\\<IMAGE_NAME>.<IMAGE_EXT>");
    await driver.executeScript("document.getElementById('readTermsOfUse').click();");
    await driver.findElement(By.name("upload_button")).submit();
    let ele = await driver.findElement(By.id("TopMessage")).getText();
    if (ele.includes('successfully uploaded')) {
      await driver.executeScript('browserstack_executor: {"action": "setSessionStatus", "arguments": {"status":"passed","reason": "File upload successful"}}');
    } else {
      await driver.executeScript('browserstack_executor: {"action": "setSessionStatus", "arguments": {"status":"failed","reason": "File upload failed"}}');
    }
  } catch (error) {
    await driver.executeScript('browserstack_executor: {"action": "setSessionStatus", "arguments": {"status":"failed","reason": "Something wrong with script"}}');
  } finally {
    await driver.quit();
  }
}
example();

File: BSTACKOPTIONS/win.cs
using OpenQA.Selenium;
using OpenQA.Selenium.IE;
using Newtonsoft.Json;
using System;

var capabilities = new DesiredCapabilities();
capabilities.SetCapability("bstack:options", JsonConvert.SerializeObject(new
    {
        os = "Windows",
        osVersion = "10",
        projectName = "Sample Test",
        buildName = "Sample_test",
        uploadMedia = new[] { "media://<FILE_HASHED_ID>" }
    }));
capabilities.SetCapability("browserName", "IE");
capabilities.SetCapability("browserVersion", "11.0");

IWebDriver driver = new RemoteWebDriver(new Uri("https://YOUR_USERNAME:YOUR_ACCESSKEY@hub-cloud.browserstack.com/wd/hub"), capabilities);

try
{
    driver.Navigate().GoToUrl("http://www.fileconvoy.com");
    IWebElement uploadElement = driver.FindElement(By.Id("upfile_0"));
    uploadElement.SendKeys("C:\\Users\\hello\\Documents\\<MEDIA_FOLDER>\\<IMAGE_NAME>.<IMAGE_EXT>");
    IJavaScriptExecutor jse = (IJavaScriptExecutor)driver;
    jse.ExecuteScript("document.getElementById('readTermsOfUse').click();");
    driver.FindElement(By.Name("upload_button")).Submit();
    IWebElement topMessage = driver.FindElement(By.Id("TopMessage"));
    if (topMessage.Text.Contains("successfully uploaded"))
    {
        jse.ExecuteScript('browserstack_executor: {"action": "setSessionStatus", "arguments": {"status":"passed","reason": "File upload successful"}}');
    }
    else
    {
        jse.ExecuteScript('browserstack_executor: {"action": "setSessionStatus", "arguments": {"status":"failed","reason": "File upload failed"}}');
    }
}
catch (Exception e)
{
    IJavaScriptExecutor jse = (IJavaScriptExecutor)driver;
    jse.ExecuteScript('browserstack_executor: {"action": "setSessionStatus", "arguments": {"status":"failed","reason": "Something wrong with script"}}');
}
finally
{
    driver.Quit();
}
